api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -75,6 +78,27 @@ public void validate(Object target, Errors errors) {
 			errors.rejectValue("stopDatetime", "Visit.error.endDateBeforeStartDate");
 		}
 		
+		try 
+		{
+			Date birthDateTime = visit.getPatient().getBirthdate();
+			Date startDateTime = visit.getStartDatetime();
+			if(startDateTime.compareTo(birthDateTime) < 0) {
+				errors.rejectValue("startDatetime","Visit.error.startDateBeforeBirthDate"); 
+			} 
+			else { 
+        			int diffInDays = (int) ((startDateTime.getTime() - birthDateTime.getTime()) / (1000 * 60 * 60 * 24));
+                		if(diffInDays<365 && diffInDays !=0) {
+                			errors.rejectValue("startDatetime","Visit.error.birthDateTooLess"); 
+                		}
+            		}
+        	}
+        	catch (NullPointerException ex) {
```

Comment:
Patient can be null sometimes and thus I'm catching the null pointer and sending an error message attribute error for the object patient